tools/shared.py

```diff
@@ -1848,6 +1848,8 @@ def link_lld(args, target, opts=[], lto_level=0):
 
     if Settings.USE_PTHREADS:
       cmd.append('--shared-memory')
+      if not Settings.WASM2JS:
```

Comment:
How about removing this, and making it error if trying to use wasm2js with pthreads? (can mention passive segments there as one thing that needs fixing). That seems more future-proof as otherwise this code silently skips adding a flag which we will need eventually.